src/v2g_exi_codec_gen/SchemaAnalyzer.py

```diff
@@ -1172,6 +1175,23 @@ def __scan_elements_for_empty_content(self):
                                 parent.has_abstract_sequence = True
                                 parent.abstract_sequences.append(abstract_seq)
 
+    def __adjust_choice_elements(self):
+        log_write('')
+        log_write('Adjusting choice elements')
+        for element in self.__generate_elements:
+            if element.has_choice:
+                choice_list = []
+                for choice in element.choices:
+                    for item in choice.choice_items:
+                        choice_list.append(item[0])
+
+                for particle in element.particles:
+                    if particle.name in choice_list:
+                        log_write(f'    Setting min_occurs of {particle.name} to 0.')
+                        particle.content_model_changed_restrictions = True
+                        particle.min_occurs_old = particle.min_occurs
+                        particle.min_occurs = 0
```

Comment:
Interesting case, I believe we do not cover child min_occur at all yet, and it is unused in our known schemas. We use the `min_occurs` variable here (and in other places) to mark the need for an `_isUsed`. The `_old` variables handle the occurrences of the overall group. In the long run, this could be changed to other flags attached to the particles, and proper handling of multiple occurrences of choices.